File: CS/ColumnSpacing/MyGridColumn.cs
using System;
using System.Linq;
using DevExpress.XtraGrid.Columns;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Base;
using System.ComponentModel;
using DevExpress.Utils.Serializing;

namespace ColumnSpacing
{
    public class MyGridColumn : GridColumn
    {
        public MyGridColumn() : base() {}

        int columnSpacing = 0;
        [Category("Appearance"),
        DefaultValue(0), XtraSerializableProperty(), Localizable(true)]
        public int ColumnSpacing {
            get {return columnSpacing;}
            set { columnSpacing = value; View.LayoutChanged(); }
        }
    }

    public class MyGridColumnCollection : GridColumnCollection
    {
        public MyGridColumnCollection(ColumnView view) : base(view) { }
        protected override GridColumn CreateColumn()
        {
            return new MyGridColumn();
        }
        public new MyGridColumn this[string fieldName] { get { return ColumnByFieldName(fieldName) as MyGridColumn; } }
        public new MyGridColumn this[int index] { get { return (MyGridColumn)List[index]; } }
    }
}

File: VB/ColumnSpacing/MyGridColumn.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports DevExpress.XtraGrid.Columns
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Views.Base
Imports System.ComponentModel
Imports DevExpress.Utils.Serializing

Namespace ColumnSpacing
	Public Class MyGridColumn
		Inherits GridColumn
		Public Sub New()
			MyBase.New()
		End Sub

		Private columnSpacing_Renamed As Integer = 0
		<Category("Appearance"), DefaultValue(0), XtraSerializableProperty(), Localizable(True)> _
		Public Property ColumnSpacing() As Integer
			Get
				Return columnSpacing_Renamed
			End Get
			Set(ByVal value As Integer)
				columnSpacing_Renamed = value
				View.LayoutChanged()
			End Set
		End Property
	End Class

	Public Class MyGridColumnCollection
		Inherits GridColumnCollection
		Public Sub New(ByVal view As ColumnView)
			MyBase.New(view)
		End Sub
		Protected Overrides Function CreateColumn() As GridColumn
			Return New MyGridColumn()
		End Function
		Default Public Shadows ReadOnly Property Item(ByVal fieldName As String) As MyGridColumn
			Get
				Return TryCast(ColumnByFieldName(fieldName), MyGridColumn)
			End Get
		End Property
		Default Public Shadows ReadOnly Property Item(ByVal index As Integer) As MyGridColumn
			Get
				Return CType(List(index), MyGridColumn)
			End Get
		End Property
	End Class
End Namespace
